plugin.audio.tripler/resources/lib/scraper.py

```diff
@@ -0,0 +1,2244 @@
+    from urlparse import parse_qs
+
+
+DATE_FORMAT = '%Y-%m-%d'
+
+URL_BASE = 'https://www.rrr.org.au'
+
+USER_AGENT = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36'
+
+stderr = False
+ignore_on_air = False
+
+
+def get(resource_path):
+    return urlopen_ua(Scraper.url_for(resource_path))
+
+def urlopen_ua(url):
+    if stderr:
+        sys.stderr.write(f"[34m# Fetching: [34;1m'{url}'[0m\n")
+    return urlopen(Request(url, headers={'User-Agent': USER_AGENT}), timeout=5)
+
+def get_json(url):
+    return urlopen_ua(url).read().decode()
+
+def get_json_obj(url):
+    return json.loads(get_json(url))
+
+def strptime(s, fmt):
+    return datetime.fromtimestamp(time.mktime(time.strptime(s, fmt)))
+
+
+class Resource:
+    def __init__(self, itemobj):
+        self._itemobj = itemobj
+
+    def id(self):
+        return self.path.split('/')[-1]
+
+    @property
+    def path(self):
+        return Scraper.resource_path_for(self._itemobj.find('a').attrs['href'])
+
+    RE_CAMEL = re.compile(r'(?<!^)(?=[A-Z])')
+    @property
+    def type(self):
+        return self.RE_CAMEL.sub('_', self.__class__.__name__).lower()
+
+    @property
+    def img(self):
+        return self._itemobj.find('img')
+
+    @property
+    def thumbnail(self):
+        if self.img:
+            return self.img.attrs.get('data-src')
+
+    @property
+    def links(self):
+        return {
+            'self': self.path
+        }
+
+    def attributes(self):
+        return {}
+
+    def links(self):
+        return {
+            'self': self.path
+        }
+
+    def relationships(self):
+        return None
+
+    def included(self):
+        return None
+
+    def to_dict(self):
+        d = {
+            'type':       self.type,
+            'id':         self.id(),
+            'attributes': {
+                'title': self.title,
+                **self.attributes(),
+            },
+            'links'     : self.links(),
+        }
+
+        r = self.relationships()
+        if r:
+            d = {
+                **d,
+                'relationships': r,
+            }
+
+        i = self.included()
+        if i:
+            d = {
+                **d,
+                'included': i,
+            }
+        return d
+
+
+
+### Scrapers ##############################################
+
+class UnmatchedResourcePath(BaseException):
+    '''
+    '''
+
+def strip_value(v):
+    if  isinstance(v, dict):
+        return strip_values(v)
+    elif isinstance(v, list):
+        return [strip_values(x) for x in v]
+    elif isinstance(v, str):
+        return v.strip()
+    else:
+        return v
+
+def strip_values(d):
+    if isinstance(d, dict):
+        return { k: remove_nulls(strip_value(v)) for k, v in d.items() }
+    else:
+        return d
+
+def remove_nulls(obj):
+    if  isinstance(obj, dict):
+        return { k: remove_nulls(v) for k, v in obj.items() if v }
```

Comment:
Good catch. I tested the difference with `if v is not None` and I'm happy with the original behavior.

We don't have any booleans in the data that this is processing, so that is fine. Along with integer `0`, this function also removes empty strings/arrays/dicts in the data, all of which is an intended effect and simplifies handling cases where the website doesn't have certain pieces of information listed properly.